x-pack/plugins/ml/public/application/components/job_spaces_list/job_spaces_list.tsx

```diff
@@ -4,20 +4,64 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React, { FC } from 'react';
+import React, { FC, useState, useEffect } from 'react';
 
-import { EuiFlexGroup, EuiFlexItem, EuiBadge } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem, EuiButtonEmpty } from '@elastic/eui';
+import { JobSpacesFlyout } from '../job_spaces_selector';
+import { JobType } from '../../../../common/types/saved_objects';
+import { useSpacesContext } from '../../contexts/spaces';
+import { Space, SpaceAvatar } from '../../../../../spaces/public';
+
+export const ALL_SPACES_ID = '*';
 
 interface Props {
-  spaces: string[];
+  spaceIds: string[];
+  jobId: string;
+  jobType: JobType;
+  refresh(): void;
+}
+
+function filterUnknownSpaces(ids: string[]) {
+  return ids.filter((id) => id !== '?');
 }
 
-export const JobSpacesList: FC<Props> = ({ spaces }) => (
-  <EuiFlexGroup wrap responsive={false} gutterSize="xs">
-    {spaces.map((space) => (
-      <EuiFlexItem grow={false} key={space}>
-        <EuiBadge color={'hollow'}>{space}</EuiBadge>
-      </EuiFlexItem>
-    ))}
-  </EuiFlexGroup>
-);
+export const JobSpacesList: FC<Props> = ({ spaceIds, jobId, jobType, refresh }) => {
```

Comment:
If you share a job to a lot of other spaces, it doesn't render too well:

<details><summary>(Screenshot)</summary>

![image](https://user-images.githubusercontent.com/5295965/99284169-ee1fb380-2803-11eb-9dc2-879b065509bb.png)

</details>

It may be worth implementing a show more / show less link like we use in the Saved Object Management page:

<details><summary>(Screenshot)</summary>

![image](https://user-images.githubusercontent.com/5295965/99284555-65554780-2804-11eb-82ed-01245ad63aa4.png)

</details>